osu.Game.Rulesets.Mania/Mods/ManiaModDoubleTime.cs

```diff
@@ -1,11 +1,14 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Game.Rulesets.Mania.Scoring;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
 
 namespace osu.Game.Rulesets.Mania.Mods
 {
-    public class ManiaModDoubleTime : ModDoubleTime
+    public class ManiaModDoubleTime : ModDoubleTime, IManiaRateAdjustmentMod
     {
+        public HitWindows HitWindows { get; set; } = new ManiaHitWindows();
```

Comment:
Not super sure about this initialisation either, but I guess the value is never used so it's probably okay... It just looks a smidge weird that this sets no-rate windows and the actual magic that undoes rate happens via the default interface implementation.

Is changing to a get-only property or a protected method something you'd be willing to entertain? Just for slightly more explicit code, is all.

Reply:
It can't be get-only if the implementation modifies it. Perhaps `CreateDefaultHitWindows()` is a possibility?

Reply:
That would work for me, I was just curious if you had any other objections (related to performance, or to the follow-up PR that does wind up/down that you talked about in the OP).

Reply:
It's still an isolated implementation, I'm fine with any if it gets the job done. We can figure out WU/WD when we cross that road.

Reply:
This is a bit more complicated to make work, so let's keep it as is for now. The likely path here is to extract the interface implementation into an auxiliary class (`HitWindowRateAdjuster`, or something like that).

It would be good to tackle this once we know the exact requirements with consideration for WU/WD.